### PR TITLE
Improved developer documentation

### PR DESCRIPTION
**Pull request by kosack**
- fixed docs for analysis module (now generated)
- added code guidelines (preliminary)
- removed future plans page, out-of-date
- started to add pull-request and code review checklists

### CODE REVIEW

File: astropy_helpers [hunk 1]
```diff
@@ -1 +1 @@
-Subproject commit 3e044eb26395c8569e3e15bf821c6ae5eeea324a
+Subproject commit 160b48587680e11f12b66a51641a2601b869d1fa
```
Comment (MaxNoe) [inline astropy_helpers]: Downgrade of the astropy-helpers. Intentional?

File: docs/development/code-guidelines.rst [hunk 2]
```diff
@@ -31,8 +35,209 @@ basic guidelines:
...
   Python 3 and new-style classes are the default, i.e. subclassing
   `object` is superfluous.
+
+
+Logging and debugging
+---------------------
+  
+* do not use `print()` statments to output text. Instead use the
+  common logging failities of `ctapipe`.  Log messages should be
+  simple, and no not include the filename, funciton name, time, or any
```
Comment (MaxNoe) [inline docs/development/code-guidelines.rst]: typo in `funciton`

File: docs/development/pullrequests.rst [hunk 3]
```diff
@@ -1,43 +1,84 @@
...
+     developer if that is normal (sometimes there are tests intended
+     to fail until a feature is available or implemented)
+   + If any other unit tests that previously suceeded now fail,
+     **stop** and ask the developer to find out what was broken by
+     their change
+     
+  - Read through the new code being contributed, and see that it
+    follows the style guidelines and that the API
+    
+   + no lines over 80 cols
```
Comment (MaxNoe) [inline docs/development/pullrequests.rst]: Most people nowadays advocate 90 characters, 80 is really restrictive.

Raymond Hattinger has a talk about this: https://www.youtube.com/watch?v=wf-BqAjZb8M

File: astropy_helpers (hunk 1 above)
Comment (kosack) [inline astropy_helpers]: no, not sure how that happened even! The old version even breaks the documentation build...  I just tried updating it but am not quite sure how to do it correctly:

```
cd astropy_helpers
git checkout v1.0.7 
```
looks fine, then

```
cd ..
git status
```
 shows nothing changed and no commit possible.  I think I just don't understand submodules. What is the correct way?

File: docs/development/pullrequests.rst (hunk 3 above)
Comment (kosack) [inline docs/development/pullrequests.rst]: I'm fine with that

**Comment by kosack** [comment]
I think this is all working now, so I will merge it (shouldn't break anything now that astropy_helpers is back to a recent version - it's going to go away shortly anyhow)

**Comment by MaxNoe** [comment]
The master was at astropy_helpers 1.2 before!

**Pull request merged**